src/api/VoteMonitor.Api.Form/Queries/UpdateFormQueryHandler.cs

```diff
@@ -0,0 +1,79 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using AutoMapper;
+using MediatR;
+using VoteMonitor.Api.Form.Models;
+using VoteMonitor.Api.Models;
+using VoteMonitor.Entities;
+
+namespace VoteMonitor.Api.Form.Queries
+{
+    public class UpdateFormQueryHandler :
+        AsyncRequestHandler<UpdateFormQuery, FormDTO>
+    {
+        private readonly VoteMonitorContext _context;
+        private readonly IMapper _mapper;
+
+        public UpdateFormQueryHandler(VoteMonitorContext context, IMapper mapper)
+        {
+            _context = context;
+            _mapper = mapper;
+        }
+
+        protected override async Task<FormDTO> HandleCore(UpdateFormQuery message)
+        {
+            var form = _context.Forms.Find(message.Id);
+
+            form.Code = message.Form.Code;
+            form.CurrentVersion = message.Form.CurrentVersion;
+            form.Description = message.Form.Description;
+            form.FormSections = new List<FormSection>();
+
+            foreach (var fs in message.Form.FormSections)
+            {
+                var formSection = MapFormSection(fs);
+                form.FormSections.Add(formSection);
+            }
+
+            await _context.SaveChangesAsync();
+            return message.Form;
+        }
+
+        private FormSection MapFormSection(FormSectionDTO formSectionDto)
```

Comment:
Can we reuse `MapFormSection` and `MapQuestion`? Maybe use a single Handler for both `AsyncRequestHandler<UpdateFormQuery, FormDTO>` and `AsyncRequestHandler<AddFormQuery, FormDTO>`  ?

Reply:
Or any other way, just so that we keep a single implementation for `MapFormSection` and `MapQuestion`

Reply:
agree on this